data-prepper-plugins/otel-trace-source/src/main/java/org/opensearch/dataprepper/plugins/source/oteltrace/OTelTraceSourceConfig.java

```diff
@@ -64,6 +65,9 @@ public class OTelTraceSourceConfig {
     @JsonProperty(ENABLE_UNFRAMED_REQUESTS)
     private boolean enableUnframedRequests = DEFAULT_ENABLED_UNFRAMED_REQUESTS;
 
+    @JsonProperty(OPENSEARCH_MODE)
+    private boolean opensearchMode = true;
```

Comment:
An enum here can help provide a clearer experience and open the door to other improvements. For example, we know we don't quite yet support simple schema.

We can have a field like `output_format` which can accept either `otel` or `opensearch`. Though I wonder if there is a better name for the OpenSearch since it is more like the original OpenSearch/Data Prepper schema.

e.g.

```
otel_trace:
  output_format: otel
```
or
```
otel_trace:
  output_format: opensearch
```

Maybe we can extend it in the future to include: `output_format: simple_schema`.

Reply:
I think `opensearch` is the correct format because if you look the existing code, every where it is clearly mentioned that the formatting in certain way for OpenSearch. For example, the following are from OTelProtoCodec.java
```
To make it OpenSearch friendly we will replace '.' in keys with '@' in all the Keys in {@link io.opentelemetry.proto.common.v1.KeyValue
```

```
Span and Resource attributes are essential for OpenSearch so they should not be nested. SO we will prefix them with "span.attributes"
     * and "resource.attributes"
 ```